AudiusProject/audius-protocol
# [PAY-3622] Update encryption patterns and add manager delegation

### Description

There were some issues with the original encryption plan, namely, we were encrypting each user's email with a symmetric key generated by the first person who purchased from a seller. This PR fixes that by going to a per user encryption basis where each user has an encrypted email and an associated encrypted key. For more details see this doc: https://www.notion.so/audiusproject/Email-Encryption-Implementation-Plan-159c00e6b203807dbe32f3edd168ea44?pvs=4

### How Has This Been Tested?

Locally on dev (must turn on own your fans feature)
 - Run local stack and generate some users with user banks and mint funds to them
 - Make a user a manager of the seller account
 - Upload a purchase gated song on one account and purchase it with the other
 - On the seller's account, download sales and ensure that you see the decrypted email
 - Do the same on the manager's account
